internal/config/flag_types.go

```diff
@@ -36,7 +36,31 @@ func (o *OutputFormat) Type() string {
 }
 
 var (
-	VarRefRx = regexp.MustCompile(`\@\{([a-zA-Z][a-zA-Z0-9_.-]*)\}`)
+	/*
+		Explanation with example:
+		Regex pattern: @{\s*(?:([a-zA-Z][a-zA-Z0-9_.-]*)\s*:)?\s*([a-zA-Z0-9_.\\/-]*)\s*}
+		String to match: "Hi! @{ embed : file.yaml } var @{ dev }!"
+
+		// Character positions (^ below represents space in the string to match)
+		// H i ! ^ @ { ^ e m b e  d  ^  :  ^  f  i  l  e  .  y  a  m  l  ^  }  ^  v  a  r  ^  @  {  ^  d  e  v  ^  }  !
+		// 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39
+
+		// Result of `match all` regex operation:
+		// [[4 26 7 12 15 24] [31 39 -1 -1 34 37]]
+
+		// Explanation:
+
+		// [4 26 7 12 15 24]               # directive with operation and operand
+		//  4 - 26: @{^embed^:^file.yaml^} (directive string)
+		//  7 - 12: embed                  (group #1: operation)
+		// 15 - 24: file.yaml              (group #2: operand)
+
+		// [31 39 -1 -1 34 37]             # plain variable substitution
+		// 31 - 39: @{^dev^}               (placeholder string)
+		// -1 - -1: NO_MATCH               (group #1: no operation to match)
+		// 34 - 37: dev                    (group #2: variable name)
+	*/
+	VarRefRx = regexp.MustCompile(`@{\s*(?:([a-zA-Z][a-zA-Z0-9_.-]*)\s*:)?\s*([a-zA-Z0-9_.\\/-]*)\s*}`)
```

Comment:
I think we can move this to the util package unexported and delete the old parsing code.